presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -100,11 +106,28 @@ public void getAllQueryInfo(
             proxyResponse(servletRequest, asyncResponse, xForwardedProto, uriInfo);
             return;
         }
+
+        int limit = firstNonNull(limitFilter, dispatchManager.getMaxQueriesToReturn());
```

Comment:
Why put this inside `DispatchManager`, rather than inject the property here directly?

Reply:
Because I don't know how to make QueryManagerConfig visible to QueryResource. :)